### PR TITLE
Clarify Vanilla Emotion auto-prefixing capabilities in documentation

### LINKED ISSUE #1523: How to get auto-prefixing with vanilla Emotion?
**Description:**

I've re-read multiple times all the different areas of the documentation that refers to vanilla Emotion usage:

https://emotion.sh/docs/introduction#framework-agnostic
https://medium.com/@tkh44/emotion-ad1c45c6d28b
https://emotion.sh/docs/install
https://emotion.sh/docs/emotion

but I'm confused about what exactly Emotion is doing with the `css`-decorated template literals and how I can get them to be auto-prefixed. In my build pipeline, I have the `autoprefixer` plugin set up with PostCSS for the CSS loader, but from what I can see, Emotion does not omit any CSS.

I think it would be nice to:
1) Describe the architecture about what emotion is doing on a high-level
2) Walk through a solution for adding vendor prefixes with vanilla emotion

### PR DESCRIPTION
**Pull request by clemmy**
Happy Thanksgiving everyone! fixes #1523

**What**:

I made this [issue](https://github.com/emotion-js/emotion/issues/1523) a while back where I had misconceptions about the capabilities of Vanilla Emotion. In this PR, I tidy up the docs a bit to hopefully erase these misconceptions (at least when read by me!).

**Why**:

These changes are needed to help Emotion users understand the capabilities of Vanilla vs React emotion better.

**How**:

Markdown changes!

**Checklist**:

- [x] Documentation
- [ ] Tests
- [ ] Code complete
- [ ] Changeset

### CODE REVIEW

File: docs/introduction.mdx [hunk 1]
```diff
@@ -63,10 +63,10 @@ The ["@emotion/core"](https://www.npmjs.com/package/@emotion/core) package requi
 
 - CSS prop support
 
-  - Similar to the `style` prop but adds support for nested selectors, media queries, and auto-prefixing.
```
Comment (Andarist) [inline docs/introduction.mdx]: I understand how lack of this information in the previous section could be confusing - but was this one here confusing as well? I would keep this section as is - mention of `style` prop is kinda nice, it indicates how it looks like in the code.

Comment (clemmy) [reply docs/introduction.mdx]: Upon re-reading this part of the code, I realize that I was associating the `style` prop with the framework agnostic section for some reason. Let me re-add this part.

File: docs/introduction.mdx [hunk 2]
```diff
@@ -61,12 +61,12 @@ The ["@emotion/core"](https://www.npmjs.com/package/@emotion/core) package requi
...
 - Best when using React with a build environment that can be configured.
 
-- CSS prop support
-
-  - Similar to the `style` prop but adds support for nested selectors, media queries, and auto-prefixing.
+- `css` prop support
 
   - Allows developers to skip the `styled` API abstraction and style components and elements directly.
 
+  - Similar to the `style` prop, but also has support for auto vendor-prefixing, nested selectors, and media queries.
```
Comment (Andarist) [inline docs/introduction.mdx]: Would prefer moving this to its former position (first in the list) - seems more important than the point about styled API.

**Comment by Andarist** [comment]
Thank you!

**Pull request merged**